src/journal/compress.c

```diff
@@ -33,6 +33,15 @@
 #include <lz4frame.h>
 #endif
 
+#if HAVE_ZSTD
+/* Yes, I know we are dynamically linking, but this is the only way to get
+ * the frame size functions.
+ */
+#define ZSTD_STATIC_LINKING_ONLY
```

Comment:
Have you considered reaching out to zstd upstream about making this better?

Reply:
hmm, let me get this right: we don't actually statically link here, right? we just define the macro to trick out the headers, and still link dynamically? I figure this is OK as long as this is reported upstream, asking them to make possible what we want to do without any such hacks...

Reply:
Please don't define this if you are not statically linking in a specific version of `libzstd`. Zstd preserves API/ABI compability between versions, except declarations hidden by `ZSTD_STATIC_LINKING_ONLY`, which have no API/ABI compatibility guarantees. It looks like the only function you are using that needs this is `ZSTD_frameHeaderSize()`, which we can remove.